### ISSUE TITLE
Annotations can initially appear in the wrong tab in long PDFs when orphans tab is enabled

### ISSUE THREAD

**Issue by robertknight** [hypothesis/client#176]
**Steps to reproduce**

1. Given an account for which the orphans_tab feature flag is enabled, activate the H client on a long PDF with many annotations
2. Open the sidebar as soon as possible and observe the Annotations and Orphans tab

Observed behavior: Many annotations initially appear in the Orphans tab and then "move" to the Annotations tab one-by-one.

**Background**

Annotations can take several seconds to anchor in long PDFs, because anchoring annotations involves rendering pages of a PDF to text which can be slow.

While annotations are anchoring, the sidebar does not display them - since it is unknown whether they should appear in the Annotations or Orphans tab. After 500ms however, annotations that are still anchoring are automatically marked (at least temporarily) as orphans. This ensures that annotations are eventually displayed in the sidebar if an error occurs during anchoring in the page (eg. due to a bug in the client or problems caused by the page's own JS or DOM structure).

On long PDFs, the result is that annotations may initially appear in the Orphans tab, then transfer one-by-one to the Annotations tab as anchoring completes.

**Possible solutions**

1. Increase the anchoring timeout if the connected document is a PDF. _This would result in annotations not appearing in the UI at all until they have been anchored, but would at least avoid them appearing in the wrong tab._
2. Display both Annotations and Orphans in the Annotations tab in PDFs to avoid them moving between tabs while anchoring completes. Orphans are still displayed with a struck-through quote to indicate that they have not been located in the document. _This would result in all annotations displaying in the Annotations tab instantly, but there would be a visual cue indicating whether an annotation can be scrolled to by clicking it_
3. Pretend that annotations always anchor in PDFs. This means always display them like an anchored annotation, even if they have not been located yet. _This would result in all annotations displaying in the Annotations tab instantly, but would provide no visual cue as to whether clicking on an annotation will scroll to it_

**Comment by dwhly** [comment]
I think 2 is a good solution.  One could also however retain the notion of a timeout, perhaps a longer one.  If time > n (5000ms?), then take all the annotations in the annotations tab that have still failed to anchor and move them to the orphans tab?

**Comment by ajpeddakotla** [comment]
@robertknight thanks for the detailed explanation here. I think that option 2 is the best option for us. Increasing the timeout might result in a frustrating user experience, so I'm not sure that we should go with that approach. 

Question here @robertknight  - once an annotation appears in the annotations tab, after anchoring is completed, can we move all non-anchored annotations to the orphans tab? Or should we stick with our original approach to treat all annotations in PDFs as [non-orphaned](https://trello.com/c/qJY6WE1t/13-always-treat-annotations-as-non-orphans-in-pdfs)?

**Comment by dwhly** [comment]
> Increasing the timeout might result in a frustrating user experience

The idea of a timeout together with 2 above is that after a period of time we silently "clean up" the unanchored annotations (if any remain) and move them over to orphans.

It specifically gets to your question "once an annotation appears in the annotations tab, after anchoring is completed, can we move all non-anchored annotations to the orphans tab?"  i.e. when is the time "after anchoring has completed"?

In other words we "treat all annotations in PDFs as non-orphaned" until the "anchoring has completed" and then move what remains to the orphans tab.

**Comment by ajpeddakotla** [comment]
Notes from planning: 
- We're mixed b/w pointing 8 or a 5
- We might split it up between 2 cards that outline 1 - the refactoring work that's needed, 2 - the client work that will distinguish between PDF/webpages

**Comment by robertknight** [comment]
Discussion of a revised and slightly simplified plan which does not require different treatments of PDFs and HTML documents: https://hypothes-is.slack.com/archives/frontend/p1482162799000721
